mk/main.mk

```diff
@@ -598,7 +598,7 @@ ALL_TARGET_RULES = $(foreach target,$(CFG_TARGET), \
 	$(foreach host,$(CFG_HOST), \
  all-target-$(target)-host-$(host)))
 
-all: $(ALL_TARGET_RULES) $(GENERATED) docs
+all: $(shell $(S)src/etc/generate_styles.py) $(ALL_TARGET_RULES) $(GENERATED) docs
```

Comment:
Target dependencies are evaluated at parse time, so this is going to cause `generate_styles.py` to be run every time make is invoked.  It should be moved to a recipe somewhere.